src/os_maild/os_maild_client.c

```diff
@@ -384,27 +384,44 @@ MailMsg *OS_RecvMailQ_JSON(file_queue *fileq, MailConfig *Mail, MailMsg **msg_sm
     os_calloc(SUBJECT_SIZE, sizeof(char), mail->subject);
 
 
-    // Get full_log field content
-    json_field = cJSON_GetObjectItem(al_json,"full_log");
-    if (!json_field) {
-        goto end;
-    }
+    /* Add alert to logs */
 
-    log_size = strlen(json_field->valuestring) + 4;
+    if(json_field = cJSON_GetObjectItem(al_json,"full_log")){
```

Comment:
Add the explicit check of the `json_field` content. Some `gcc` versions complain about this:

```
os_maild/os_maild_client.c: In function ‘OS_RecvMailQ_JSON’:
os_maild/os_maild_client.c:389:8: warning: suggest parentheses around assignment used as truth value [-Wparentheses]
     if(json_field = cJSON_GetObjectItem(al_json,"full_log")){
        ^~~~~~~~~~
```

By adding:

```
if(json_field = cJSON_GetObjectItem(al_json,"full_log"), json_field){
```
we avoid the error